### ISSUE TITLE
[$250]  Profile - Zoom slider does not change position when moving to zoom an image

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#45164]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.6-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** gatantm+528@gmail.com
**Issue reported by:** Applause - Internal Team

**Issue found when executing PR** https://github.com/Expensify/App/pull/44788
## Action Performed:
1. Navigate to account Settings > Profile
2. Click on the avatar
3. Click on upload photo and upload any JPEG image
4. Hover over the zoom slider and move to the right

## Expected Result:
The zoom slider is moving to show how an image is zoomed

## Actual Result:
The zoom slider stays on the same position when moving it to zoom uploaded image

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/fd7a8c59-f557-4379-882d-2f15fd7edbd0

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0130ac939be52a2a80</li>
        <li>Upwork Job ID: 1811335809081764317</li>
        <li>Last Price Increase: 2024-09-19</li>
    </ul>
</details>

**Comment by lanitochka17** [comment]
@Julesssss  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by Julesssss** [comment]
Cute cat.

This needs to be fixed, but I'm demoting it from being a blocker.

**Comment by Julesssss** [comment]
My best guess so far is that [this PR](https://github.com/Expensify/App/pull/44788) introduced the bug

**Comment by Julesssss** [comment]
Thanks for looking

**Comment by harshit078** [comment]
On reproducing the issue, Zoom slider does change its position but it's only when clicked and doesnt move when dragged. No animation for dragging

https://github.com/Expensify/App/assets/73997189/9dac4f95-5bd7-41ad-ae19-3e9eedf6eee6

**Comment by QichenZhu** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The zoom slider doesn't work.

### What is the root cause of that problem?
In versions below `2.16.0`, `react-native-gesture-handler` registers `pointermove` listeners but never unregisters them. This causes issues in development mode since `StrictMode` renders components twice. This has been fixed by https://github.com/software-mansion/react-native-gesture-handler/pull/2815.

### What changes do you think we should make in order to solve the problem?
1. Wait until #45162 is resolved. Currently I'm using the work-in-progress solution from #45314.
2. Upgrade `react-native-gesture-handler` to version `2.16.0` or newer.

### What alternative solutions did you explore? (Optional)
N/A

**Comment by hungvu193** [comment]
I'll take a look this weekend

**Comment by Ahmed-Abdella** [comment]
Please check my [Proposal](https://github.com/Expensify/App/issues/45162#issuecomment-2227062344) here, it fixes the Crash when trying to upload the Avatar. I believe it fixes this issue also (maybe not completely).

**Comment by hungvu193** [comment]
> Please check my [Proposal](https://github.com/Expensify/App/issues/45162#issuecomment-2227062344) here, it fixes the Crash when trying to upload the Avatar. I believe it fixes this issue also.

Your proposal did fixed the crashes but didn't fix this issue, also in your RCA's proposal, you mentioned the code that was added 14 months ago, I don't think that's correct RCA.

**Comment by hungvu193** [comment]
@QichenZhu Your proposal looks promising but I'm facing a crash while testing it on mobile. Did you test it with mobile platform?

**Comment by QichenZhu** [comment]
@hungvu193 Yes, I came across the same problem on mobile, so I suggested holding for #45162 for now.

**Comment by hungvu193** [comment]
> @hungvu193 Yes, I came across the same problem on mobile, so I suggested holding for #45162 for now.

I actually applied your proposal from that issue along with your mentioned PR, but it still crashed

**Comment by QichenZhu** [comment]
@hungvu193 Thanks for testing and pointing out the problem. I'll revisit #45162 in a few hours.

**Comment by alexpensify** [comment]
I think that this notice is an error, but correct me if I'm wrong @hungvu193. Thanks!

**Comment by hungvu193** [comment]
> I think that this notice is an error, but correct me if I'm wrong @hungvu193. Thanks! 

Yeah. This issue has similar root cause with a DB, so a contributor mentioned in DB that his proposal will fix this issue too that's why we saw this alert. We can ignore that 😄

**Comment by QichenZhu** [comment]
@hungvu193 The fix for #45162 has been merged. Could you retest on the latest `main`?

**Comment by hungvu193** [comment]
> @hungvu193 The fix for #45162 has been merged. Could you retest on the latest `main`?

Sure. I'll retest it tomorrow in my time.

**Comment by QichenZhu** [comment]
@hungvu193 I am withdrawing my [proposal](https://github.com/Expensify/App/issues/45164#issuecomment-2225980133) because `react-native-gesture-handler` `2.16.0`, despite its [README](https://github.com/software-mansion/react-native-gesture-handler/blob/e4e23863562df69037cc03bc46c968aa010f74d3/README.md?plain=1#L40) claiming to support React Native `0.68.0+`, does not support React Native `0.73.4` as mentioned in [this issue](https://github.com/software-mansion/react-native-gesture-handler/issues/2927):

> Our current policy regarding the new architecture support is to support only the most recent version, which at the time of writing is 0.74. The crash you're describing is caused by a breaking change between RN 0.73 and 0.74.

**Comment by hungvu193** [comment]
@QichenZhu Thanks for your update!
   

Still waiting for proposals.

**Comment by hungvu193** [comment]
I re-test it today, I think this issue is fixed.

https://github.com/user-attachments/assets/8276fb23-f321-4b0b-9289-10dfe8c03a52

**Comment by kavimuru** [comment]
Bug is fixed

https://github.com/user-attachments/assets/25783ca4-96f8-4d19-829a-39430bfb67fd

**Comment by Ahmed-Abdella** [comment]
Why I am still able to reproduce it on the latest main v9.0.8-2


https://github.com/user-attachments/assets/003acbf5-e1da-459d-a858-90be8717c0d1

**Comment by QichenZhu** [comment]
@hungvu193 @kavimuru @Ahmed-Abdella The bug is reproducible in `development` mode, but not in `production` mode.

#### Development mode

https://github.com/user-attachments/assets/53c96227-6f4e-4d3f-8ca8-01f92f89138d


#### Production mode

https://github.com/user-attachments/assets/52fc381e-d371-43ad-b29a-a32195c360ac

**Comment by hungvu193** [comment]
Oops I think I disabled StrictMode during test this issue

**Comment by Ahmed-Abdella** [comment]
## Proposal
 ### Please re-state the problem that we are trying to solve in this issue.
 The zoom slider doesn't work.
 
### What is the root cause of that problem?

in the slider component

https://github.com/Expensify/App/blob/7f66ea3632c787bce4ab8b6dc12c252c1ddc7685/src/components/AvatarCropModal/Slider.tsx#L41-L53

the tooltipIsVisible state doesn't change in a correct way Here is what happened

the minDistance(5) make sure that the gesture doesn't get active until after 5 finger 

>minDistance(value: number)
Minimum distance the finger (or multiple finger) need to travel before the gesture [activates](https://docs.swmansion.com/react-native-gesture-handler/docs/fundamentals/states-events#active). Expressed in points.

so the tooltipIsVisible state change to true before activating the gesture which prevent us to control the gesture handler

 
### What changes do you think we should make in order to solve the problem?


We should call the `minDistance()` method with 0 instead of 5

```js
Gesture.Pan()
        .minDistance(0)
```     

by change the value to 0 we ensure the gesture active immediately   


### Result

https://github.com/user-attachments/assets/6f1c64b1-386b-47b6-971c-b4a1356bb922






### What alternative solutions did you explore? (Optional)
 N/A

**Comment by Ahmed-Abdella** [comment]
Proposal updated for more explanation.

**Comment by hungvu193** [comment]
Thanks for your update. I'll review it on my morning.

**Comment by hungvu193** [comment]
> So we lose the control of gesture handler before the activation, and now we can't active the gesture anymore and the `onChange` method (event) that's responsible to update the sllider will never run.

Hey can you give me more detailed about this one right here?

Also please notice that this issue is now only reproduced when StrictMode is enabled.

**Comment by Ahmed-Abdella** [comment]
> Hey can you give me more detailed about this one right here?

Maybe I complicated things a little bit, all I mean is that the `onChange` callback in `Gesture.pan()` never got invoked because we delayd the activation for a distance of 5 by using `minDistance(5)`.

the `onBegin` callback got invoke, and immediately after start moving the `onFinalize` got invoked, means that the gesture ended. And the `onChange` callback is skipped. So we need to active the gesture immediately to trigger the `onChange` callback and apply the updates inside it.

> Also please notice that this issue is now only reproduced when StrictMode is enabled.

Yes, Maybe because the fact that the StrictMode double invoke the component to identify the side effects and also enforce synchronous state updates that cause immediate rerender.

Maybe this is causing problems in how the state got managed and updated in the gesture handler after Delaying the activation using `minDistance`, which make the onChange not being invoked and the onFinalize called before it's time.

Setting the minDistance to 0 ensure that the gesture activate immediatey and the `onChange` that update the slider got invoked.

**Comment by hungvu193** [comment]
I've been under the water for past few days, I'll take another look today

**Comment by hungvu193** [comment]
Reviewing shortly

**Comment by hungvu193** [comment]
> Maybe I complicated things a little bit, all I mean is that the `onChange` callback in `Gesture.pan()` never got invoked because we delayd the activation for a distance of 5 by using `minDistance(5)`.

I don't think this is correct and setting `minDistance` to 0 feels like a workaround to me. 


Upgrade `react-native-gesture-handler` will fix this issue, however we already had a PR to do that:
https://github.com/Expensify/App/pull/40548

@alexpensify Please put this PR on hold for https://github.com/Expensify/App/pull/40548, this bug's only reproduced on StrictMode, so I think that's fine to hold it.

**Comment by alexpensify** [comment]
Thanks for the update! I've updated this GH.

**Comment by alexpensify** [comment]
Weekly Update: On hold for https://github.com/Expensify/App/pull/40548

**Comment by alexpensify** [comment]
Weekly Update: hold for https://github.com/Expensify/App/pull/40548

**Comment by alexpensify** [comment]
Weekly Update:  https://github.com/Expensify/App/pull/40548

**Comment by alexpensify** [comment]
Weekly update: The other GH we have been tracking states - `Closing this in favor of the 0.75 upgrade` 

I need to figure out what we should do here. @roryabraham any feedback? Thanks!

**Comment by roryabraham** [comment]
Should HOLD on https://github.com/Expensify/App/issues/37374 instead

**Comment by alexpensify** [comment]
Weekly Update: Thanks, Rory! Still on hold for https://github.com/Expensify/App/issues/37374

**Comment by roryabraham** [comment]
We are on RN 0.75 now, so I think this can come off hold. That issue isn't closed yet but is in the cleanup phase.

**Comment by alexpensify** [comment]
Thank you, @roryabraham, for the update. @hungvu193 please keep us posted if you agree with this plan. Thanks!

**Comment by hungvu193** [comment]
@alexpensify I think this is fixed on latest main, can you re-test this issue?

https://github.com/user-attachments/assets/c9b178f7-5512-43eb-9329-e4a78fa0b636

**Comment by alexpensify** [comment]
@lanitochka17 can your team please retest to confirm if there is still an issue? Thanks!

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by alexpensify** [comment]
Weekly update: waiting on a retest

**Comment by alexpensify** [comment]
Weekly Update: I'll need to ask in Slack soon if no movement here.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by alexpensify** [comment]
Closing - this is confirmed in the recent test.
